runtime/src/accounts.rs

```diff
@@ -133,15 +134,24 @@ impl Accounts {
             if accounts.is_empty() || accounts[0].lamports == 0 {
                 error_counters.account_not_found += 1;
                 Err(TransactionError::AccountNotFound)
-            } else if accounts[0].owner != system_program::id() {
-                error_counters.invalid_account_for_fee += 1;
-                Err(TransactionError::InvalidAccountForFee)
-            } else if accounts[0].lamports < fee {
-                error_counters.insufficient_funds += 1;
-                Err(TransactionError::InsufficientFundsForFee)
             } else {
-                accounts[0].lamports -= fee;
-                Ok((accounts, tx_rent))
+                let min_balance = match account_is_system(&accounts[0]).ok_or_else(|| {
+                    error_counters.invalid_account_for_fee += 1;
+                    TransactionError::InvalidAccountForFee
+                })? {
+                    SystemAccountKind::System => 0,
+                    SystemAccountKind::Nonce => {
+                        rent_collector.rent.minimum_balance(NonceState::size())
+                    }
+                };
+
+                if accounts[0].lamports < fee + min_balance && fee != accounts[0].lamports {
```

Comment:
full withdrawal is not supported here, the Nonce must live beyond this TX if it is a Nonce...

Reply:
Right, duh!  Only close thorough `NonceWithdraw`... same for `bank::withdraw()`.

Reply:
https://github.com/solana-labs/solana/pull/7645/commits/35fa57c17c4bdb4367613e1256688cbbc97b4c87